rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -875,39 +953,64 @@ void PlayerImpl::prepare_publishers()
   auto topics = reader_->get_all_topics_and_types();
   std::string topic_without_support_acked;
   for (const auto & topic : topics) {
-    if (publishers_.find(topic.name) != publishers_.end()) {
+    if (senders_.find(topic.name) != senders_.end()) {
       continue;
     }
-    // filter topics to add publishers if necessary
+
     auto & filter_topics = storage_filter.topics;
-    if (!filter_topics.empty()) {
-      auto iter = std::find(filter_topics.begin(), filter_topics.end(), topic.name);
-      if (iter == filter_topics.end()) {
-        continue;
+    auto & filter_services = storage_filter.services;
+
+    if (rosbag2_cpp::is_service_event_topic(topic.name, topic.type)) {
+      // filter services to add clients if necessary
+      if (!filter_services.empty()) {
+        auto iter = std::find(filter_services.begin(), filter_services.end(), topic.name);
+        if (iter == filter_services.end()) {
+          continue;
+        }
       }
-    }
-
-    auto topic_qos = publisher_qos_for_topic(
-      topic, topic_qos_profile_overrides_,
-      owner_->get_logger());
-    try {
-      std::shared_ptr<rclcpp::GenericPublisher> pub =
-        owner_->create_generic_publisher(topic.name, topic.type, topic_qos);
-      std::shared_ptr<PlayerImpl::PlayerPublisher> player_pub =
-        std::make_shared<PlayerImpl::PlayerPublisher>(
-        std::move(pub), play_options_.disable_loan_message);
-      publishers_.insert(std::make_pair(topic.name, player_pub));
-      if (play_options_.wait_acked_timeout >= 0 &&
-        topic_qos.reliability() == rclcpp::ReliabilityPolicy::BestEffort)
-      {
-        topic_without_support_acked += topic.name + ", ";
+      auto service_name = rosbag2_cpp::service_event_topic_name_to_service_name(topic.name);
+      auto service_type = rosbag2_cpp::service_event_topic_type_to_service_type(topic.type);
+      try {
+        auto cli = owner_->create_generic_client(service_name, service_type);
+        auto player_cli = std::make_shared<PlayerClient>(
+          std::move(cli), service_name, topic.type, owner_->get_logger());
+        senders_.insert(std::make_pair(topic.name, player_cli));
+      } catch (const std::runtime_error & e) {
+        RCLCPP_WARN(
+          owner_->get_logger(),
+          "Ignoring a service '%s', reason: %s.",
+          service_name.c_str(), e.what());
```

Comment:
Comments from MichaelOrlov

```suggestion
        RCLCPP_WARN(
          owner_->get_logger(),
          "Ignoring a service '%s', reason: %s.", service_name.c_str(), e.what());
```
Nitpick

Reply:
Updated